### PR TITLE
Email marketing

### PR DESCRIPTION
**Pull request by kempsterrrr**
## Describe your changes

- Created custom `/verify-request` page
- introduce local email templates

No DB changes. No analytics events.

## Issue ticket number and link

NA

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I've included any relevant links to DB changes/deploy-requests
- [x] I've added any required Analytics events

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v16.15.1
+v18.12.0
```
Comment (mbuxmann) [inline .nvmrc:1]: I assume since we aren't using postmark sdk anymore we bumping this up to 18 again?

File: src/pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -29,11 +29,172 @@ export const authOptions: NextAuthOptions = {
...
+    verifyRequest: "/auth/verify-request",
   },
 };
 
+async function CustomsendVerificationRequest(params: {
+  identifier: any;
+  url: any;
+  provider: any;
+  theme?: any;
+}) {
```
Comment (mbuxmann) [inline src/pages/api/auth/[...nextauth].ts:48]:
```
async function CustomSendVerificationRequest(params: {
  identifier: string;
  url: string;
  provider: any;
})
```

Comment (mbuxmann) [reply src/pages/api/auth/[...nextauth].ts:48]: I'm sure a provider has a type we can find. We should try and avoid any at all costs.

File: src/pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -29,11 +29,172 @@ export const authOptions: NextAuthOptions = {
...
+  const { identifier, url, provider, theme } = params;
```
Comment (mbuxmann) [inline src/pages/api/auth/[...nextauth].ts:49]: theme can be removed

File: src/pages/api/auth/[...nextauth].ts [hunk 4]
```diff
@@ -29,11 +29,172 @@ export const authOptions: NextAuthOptions = {
...
+    text: text({ url, host }),
+    html: html({ url, host, theme }),
+  });
+  const failed = result.rejected.concat(result.pending).filter(Boolean);
+  if (failed.length) {
+    throw new Error(`Email(s) (${failed.join(", ")}) could not be sent`);
+  }
+}
+
+function html(params: { url: string; host: string; theme: Theme }) {
```
Comment (mbuxmann) [inline src/pages/api/auth/[...nextauth].ts:66]: remove theme

File: src/pages/api/auth/[...nextauth].ts [hunk 5]
```diff
@@ -29,11 +29,172 @@ export const authOptions: NextAuthOptions = {
...
+  const { url, host, theme } = params;
```
Comment (mbuxmann) [inline src/pages/api/auth/[...nextauth].ts:67]: remove theme

**Review by mbuxmann — CHANGES_REQUESTED** [review]

File: src/pages/api/auth/[...nextauth].ts (hunk 2 above)
Comment (mbuxmann) [reply src/pages/api/auth/[...nextauth].ts:48]: if not just add a note so we can look into it later. 

`// TODO(refactor): Add provider type`

**Review by mbuxmann — APPROVED** [review]
LGTM, just some recommended changes 👍🏻

File: src/pages/api/auth/[...nextauth].ts (hunk 2 above)
Comment (kempsterrrr) [reply src/pages/api/auth/[...nextauth].ts:48]: Found it ! EmailConfig

Should the URL type be URL?

**Comment by kempsterrrr** [comment]
Update, let one comment regarding URL vs String type

**Comment by kempsterrrr** [comment]
Oooops. Think I might have fucked up by merging in staging locally 🤦🏻 cc @mbuxmann

File: src/pages/api/auth/[...nextauth].ts (hunk 2 above)
Comment (mbuxmann) [reply src/pages/api/auth/[...nextauth].ts]: no string is fine. it seems like it does convert it for you to a URL below `const { host } = new URL(url);`

Comment (kempsterrrr) [reply src/pages/api/auth/[...nextauth].ts]: awesome, thanks

File: .nvmrc (hunk 1 above)
Comment (kempsterrrr) [reply .nvmrc:1]: Yup works fine for me

**Pull request merged**